internal/oidc/provider.go

```diff
@@ -1,8 +1,59 @@
 package oidc
 
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+	"strings"
+)
+
 type Provider struct {
 	IssuerURL string
 	Blocked   bool
 	JWKSURIs  []string
 	Audiences []string
 }
+
+type TokenSet struct {
+	AccessToken  string `json:"access_token"`
+	RefreshToken string `json:"refresh_token"`
+	IDToken      string `json:"id_token"`
+	TokenType    string `json:"token_type"`
+	ExpiresIn    int    `json:"expires_in"`
+}
+
+func (p *Provider) Exchange(ctx context.Context, code, codeVerifier, redirectURI string, clientID string) (*TokenSet, error) {
```

Comment:
The `oidc.Provider` type is designed to be a value object. It should not have `Exchange` method which executes an HTTP request. Rather, this method should be a part of some domain service. This enables extensibility.

https://martinfowler.com/bliki/ValueObject.html

https://www.ompluscator.com/article/golang/practical-ddd-value-object/
https://www.ompluscator.com/article/golang/practical-ddd-entity/
https://www.ompluscator.com/article/golang/practical-ddd-domain-service/

https://github.com/openkcm/session-manager/pull/22#discussion_r2410582371

If you pull the latest changes from main, there's an `oidc.Service` type, which is the right place to put such a method.